src/OrchardCore.Modules/OrchardCore.Html/Handlers/HtmlBodyPartHandler.cs

```diff
@@ -25,12 +27,17 @@ public override Task GetContentItemAspectAsync(ContentItemAspectContext context,
         {
             return context.ForAsync<BodyAspect>(async bodyAspect =>
             {
-                if (_bodyAspect != null)
+                if (bodyAspect != null && part.ContentItem.Id == _contentItemId)
                 {
                     bodyAspect.Body = _bodyAspect;
+                    part.ContentItem.Id = _contentItemId;
```

Comment:
why are you setting this here? You've already compared it as equal, so no need to set?